code/modules/mob/living/carbon/human/species_types/psyphoza.dm

```diff
@@ -0,0 +1,362 @@
+		return FALSE
+	return ..()
+
+/datum/species/psyphoza/spec_life(mob/living/carbon/human/H)
+	if(world.time % 3)
+		return
+	var/obj/effect/psyphoza_spores/P = locate(/obj/effect/psyphoza_spores) in H.loc
+	if(P && H.health < H.maxHealth)
+		//healing
+		var/amount = P.take_resources(3 * P.upgrades)
+		H.heal_overall_damage(amount,amount, 0, BODYTYPE_ORGANIC)
+
+/datum/species/psyphoza/get_scream_sound(mob/living/carbon/user)
+	return pick('sound/voice/psyphoza/psyphoza_scream_1.ogg', 'sound/voice/psyphoza/psyphoza_scream_2.ogg')
+
+//This originally held the psychic action until I moved it to the eyes, keep it please.
+/obj/item/organ/brain/psyphoza
+	name = "psyphoza brain"
+	desc = "Bubbling with psychic energy..no wait...that's blood."
+	color = "#ff00ee"
+
+// PSYCHIC ECHOLOCATION
+/datum/action/item_action/organ_action/psychic_highlight
+	name = "Psychic Sense"
+	desc = "Sense your surroundings psychically."
+	///The distant our psychic sense works
+	var/sense_range = 5
+	///The range we can hear-ping things from
+	var/hear_range = 8
+	///List of things we can't sense
+	var/list/sense_blacklist
+	///The amount of time you can sense things for
+	var/sense_time = 7 SECONDS
+	///Reference to the users eyes - we use this to toggle xray vision for scans
+	var/obj/item/organ/eyes/eyes
+	///The eyes original sight flags - used between toggles
+	var/sight_flags
+	///Time between uses
+	var/cooldown = 1 SECONDS
+	///List of overlays we made
+	var/list/overlays = list()
+	///Reference to 'kill these overlays' timer
+	var/overlay_timer
+
+/datum/action/item_action/organ_action/psychic_highlight/New(Target)
+	. = ..()
+	//Setup massive blacklist typecache of non-renderables. Smaller than whitelist
+	sense_blacklist = typecacheof(GLOB.psychic_sense_blacklist)
+
+/datum/action/item_action/organ_action/psychic_highlight/Grant(mob/M)
+	. = ..()
+	//Register signal for TK highlights
+	RegisterSignal(M, COMSIG_MOB_ATTACK_RANGED, .proc/handle_ranged)
+	//Overlay used to highlight objects
+	M.overlay_fullscreen("psychic_highlight", /atom/movable/screen/fullscreen/blind/psychic_highlight)
+	//Add option to change visuals
+	var/datum/action/change_psychic_visual/P = new()
+	P.Grant(owner)
+
+/datum/action/item_action/organ_action/psychic_highlight/Trigger()
+	. = ..()
+	if(has_cooldown_timer || !owner)
+		return
+	ping_turf(get_turf(owner))
+	has_cooldown_timer = TRUE
+	addtimer(CALLBACK(src, .proc/finish_cooldown), cooldown + (sense_time * min(1, overlays.len / PSYCHIC_OVERLAY_UPPER)))
+
+/datum/action/item_action/organ_action/psychic_highlight/proc/finish_cooldown()
+	has_cooldown_timer = FALSE
+
+//Allows user to see images through walls
+/datum/action/item_action/organ_action/psychic_highlight/proc/toggle_eyes_fowards()
+	//Grab organs - we do this here becuase of fuckery :tm:
+	if(!eyes && istype(owner, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = owner
+		//eyes
+		eyes = eyes || locate(/obj/item/organ/eyes) in H.internal_organs
```

Comment:
```suggestion
		eyes = locate(/obj/item/organ/eyes) in H.internal_organs
```
eyes is always null here

Reply:
``eyes`` is null at the start then assigned from internal organs. When those eyes are removed from the body ``eyes`` will be nulled. Otherwise ``eyes`` won't be null, having been assigned.  By checking, we can avoid calling ``locate`` every activation.